pkg/controllers/node/termination/terminator/suite_test.go

```diff
@@ -128,53 +129,43 @@ var _ = Describe("Eviction/Queue", func() {
 				Labels:         testLabels,
 				MaxUnavailable: &intstr.IntOrString{IntVal: 1},
 			})
-			ExpectApplied(ctx, env.Client, pod)
-			Expect(queue.Evict(ctx, terminator.NewQueueKey(pod, node.Spec.ProviderID))).To(BeTrue())
+			ExpectApplied(ctx, env.Client, pod, node)
+			Expect(queue.Evict(ctx, pod)).To(BeNil())
 			Expect(recorder.Calls(events.Evicted)).To(Equal(1))
 		})
 		It("should return a NodeDrainError event when a PDB is blocking", func() {
-			ExpectApplied(ctx, env.Client, pdb, pod)
-			Expect(queue.Evict(ctx, terminator.NewQueueKey(pod, node.Spec.ProviderID))).To(BeFalse())
+			ExpectApplied(ctx, env.Client, pdb, pod, node)
+			Expect(queue.Evict(ctx, pod)).ToNot(Succeed())
 			Expect(recorder.Calls(events.FailedDraining)).To(Equal(1))
 		})
 		It("should fail when two PDBs refer to the same pod", func() {
 			pdb2 := test.PodDisruptionBudget(test.PDBOptions{
 				Labels:         testLabels,
 				MaxUnavailable: &intstr.IntOrString{IntVal: 0},
 			})
-			ExpectApplied(ctx, env.Client, pdb, pdb2, pod)
-			Expect(queue.Evict(ctx, terminator.NewQueueKey(pod, node.Spec.ProviderID))).To(BeFalse())
+			ExpectApplied(ctx, env.Client, pdb, pdb2, pod, node)
+			Expect(queue.Evict(ctx, pod)).ToNot(Succeed())
 			ExpectMetricCounterValue(terminator.NodesEvictionRequestsTotal, 1, map[string]string{terminator.CodeLabel: "500"})
 		})
 		It("should ensure that calling Evict() is valid while making Add() calls", func() {
-			cancelCtx, cancel := context.WithCancel(ctx)
-			wg := sync.WaitGroup{}
-			DeferCleanup(func() {
-				cancel()
-				wg.Wait() // Ensure that we wait for reconcile loop to finish so that we don't get a RACE
-			})
-
-			// Keep calling Reconcile() for the entirety of this test
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
+			// Ensure that we add enough pods to the queue while we are pulling items off of the queue (enough to trigger a DATA RACE)
+			pods := test.Pods(1000)
 
-				for {
-					ExpectSingletonReconciled(ctx, queue)
-					if cancelCtx.Err() != nil {
-						return
+			for _, pod := range pods {
+				go func() {
```

Comment:
These goroutines are never canceled, so when they start up, they are going to continue to operate and run for the lifetime of the test. I think you need to either have them attempt to add the pod to the queue a single time and then exit or you build in cancelation with `context.WithCancel()` and cancel the new context that you build at the end of the function, also causing the goroutines to bail